### PR TITLE
Slay the Dragon!

### PR DESCRIPTION
**Pull request by diasdavid**
It seems that **running our tests in Node.js 8 always fail in AppVeyor and locally** (not in Travis or Circle CI interestingly enough)

### CODE REVIEW

File: test/index.spec.js [hunk 1]
```diff
@@ -467,34 +467,32 @@ describe('daemons', () => {
...
 
     it('starts the daemon and returns valid API and gateway addresses', (done) => {
-      let daemon
       const dir = `${os.tmpdir()}/tmp-${Date.now() + '-' + Math.random().toString(36)}`
 
       async.waterfall([
         (cb) => ipfsd.local(dir, cb),
-        (node, cb) => {
-          daemon = node
-          node.init((err) => cb(err, node))
```
Comment (diasdavid) [inline test/index.spec.js]: Seems like in Node.js 8 a copy was being made instead of a reference.. WAT?

Comment (dignifiedquire) [reply test/index.spec.js]: Hmm that sounds dubious, we should change the test, if this is making copies there is likely a larger issue at play which will be a problem for users of ipfsd-ctl

Comment (diasdavid) [reply test/index.spec.js]: It seems that this didn't fixed it after all, I just got lucky

Comment (diasdavid) [reply test/index.spec.js]: got it

**Comment by diasdavid** [comment]
This PR solves the issues that https://github.com/ipfs/js-ipfsd-ctl/pull/150 was trying to solve.

**Comment by diasdavid** [comment]
Given the _sensitivity_ of this module, I appreciate all the CR I can get :)

tl;dr; There were some big issues on how daemons were being spawned and it mostly worked because the race condition was in our favor. Node.js 8 changed that.

**Comment by diasdavid** [comment]
image

wubbalubbadubdub 🎉

File: test/spawning-daemons.spec.js [hunk 2]
```diff
@@ -467,34 +411,32 @@ describe('daemons', () => {
...
-        expect(res).to.have.property('gatewayPort')
-        expect(res.apiHost).to.equal('127.0.0.1')
-        expect(res.apiPort).to.equal('5001')
-        expect(res.gatewayHost).to.equal('127.0.0.1')
-        expect(res.gatewayPort).to.equal('8080')
+        // Check for props in ipfs-api instance
+        expect(api).to.have.property('apiHost')
+        expect(api).to.have.property('apiPort')
+        expect(api).to.have.property('gatewayHost')
+        expect(api).to.have.property('gatewayPort')
```
Comment (VictorBjelkholm) [inline test/spawning-daemons.spec.js]: These checks are unnecessary as the ones below will check if the property exists and is equal to the value.

File: src/index.js [hunk 3]
```diff
@@ -109,6 +91,29 @@ const IpfsDaemonController = {
     } else {
       node.init((err) => callback(err, node))
     }
+  },
+
+  /**
+   * Create a new disposable node and already start the daemon.
```
Comment (VictorBjelkholm) [inline src/index.js]: Minor typo
> already started daemon

**Comment by diasdavid** [comment]
Thanks @VictorBjelkholm :)
